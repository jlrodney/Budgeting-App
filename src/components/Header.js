import React from 'react';
import { NavLink } from 'react-router-dom';
import {connect} from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header>
    <h1 className="header">Budgeting App</h1>
    <div className="option">
      <NavLink className="home" to="/homepage" activeClassName="is-active" >Homepage</NavLink>
      <NavLink to="/create" className="create" activeClassName="is-active">Create Expense</NavLink>
      <NavLink to="/help" className="help" activeClassName="is-active" exact={true}>Help</NavLink>
      <button onClick={startLogout}>Logout</button>
    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
