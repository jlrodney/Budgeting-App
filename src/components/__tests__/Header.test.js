import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { Header } from '../Header';

describe('Header should render  correctly', () => {
  it('should have headline', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('h1').text()).toEqual("Budgeting App");
  })
  it('should have a link to homepage', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find("NavLink").length).toBe(3)
    expect(wrapper.find('NavLink.home').prop('to')).toBe("/homepage");
    expect(wrapper.find('NavLink.create').prop('to')).toBe("/create");
    expect(wrapper.find('NavLink.help').prop('to')).toBe("/help");
  })

  it('should call logout on button click', () => {
    const startLogOutSpy = jest.fn();
    const wrapper = shallow(<Header startLogout={startLogOutSpy} />);
    wrapper.find('button').simulate('click');
    expect(startLogOutSpy).toHaveBeenCalled();
  })
});

describe('should render Header correctly', () => {
  const wrapper = shallow(<Header startLogout={() => { }} />);
  expect(wrapper).toMatchSnapshot();
})
