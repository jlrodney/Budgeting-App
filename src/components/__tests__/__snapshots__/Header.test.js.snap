// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<header>
  <h1
    className="header"
  >
    Budgeting App
  </h1>
  <div
    className="option"
  >
    <NavLink
      activeClassName="is-active"
      aria-current="page"
      className="home"
      to="/homepage"
    >
      Homepage
    </NavLink>
    <NavLink
      activeClassName="is-active"
      aria-current="page"
      className="create"
      to="/create"
    >
      Create Expense
    </NavLink>
    <NavLink
      activeClassName="is-active"
      aria-current="page"
      className="help"
      exact={true}
      to="/help"
    >
      Help
    </NavLink>
    <button
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</header>
`;
